### PR TITLE
refactor(datetime): update ionic styles for footer buttons

### PR DESCRIPTION
**Pull request by thetaPC**
Issue number: internal

---------

## What is the current behavior?

The footer buttons (confirm, clear, and cancel) do not match the ionic theme styles.

## What is the new behavior?

- Update the styles to mimic the design.
- Footer buttons render full width when the theme is `ionic`.
- The confirm button renders as a solid fill when the theme is `ionic`.
- Added snapshots.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

[Preview](https://ionic-framework-git-rou-11363-ionic1.vercel.app/src/components/datetime/test/month-year-picker?ionic:theme=ionic)

### CODE REVIEW

File: core/src/components/buttons/buttons.ionic.scss [hunk 1]
```diff
@@ -13,7 +13,7 @@
 // --------------------------------------------------
 
 ::slotted(*) .button-clear {
-  --color: globals.$ion-primitives-neutral-1200;
+  --color: #{globals.$ion-primitives-neutral-1200};
```
Comment (thetaPC) [inline core/src/components/buttons/buttons.ionic.scss:16]: Noticed this while updating the styles.

File: core/src/components/datetime/datetime.common.scss [hunk 2]
```diff
@@ -153,30 +153,15 @@
...
- * The confirm and clear buttons are grouped in a
- * container so that they appear on the end opposite
- * of the cancel button.
- * We use display: flex so that the
- * wrapper only takes up as much
- * height as it needs.
- */
-.datetime-action-buttons .datetime-action-buttons-container {
-  display: flex;
-}
```
Comment (thetaPC) [inline core/src/components/datetime/datetime.common.scss:179]: These only apply to the native styles.

File: core/src/components/datetime/datetime.ionic.scss [hunk 3]
```diff
@@ -200,3 +200,16 @@
   background-color: globals.$ion-primitives-neutral-100;
   color: globals.$ion-primitives-neutral-1200;
 }
+
+// Calendar / Footer / Action Buttons
+// -----------------------------------
+
+:host .datetime-buttons ion-buttons {
+  display: block;
```
Comment (brandyscarney) [inline core/src/components/datetime/datetime.ionic.scss:208]:
```suggestion
  display: flex;
  flex-flow: column;
  align-items: stretch;
  gap: globals.$ion-space-250;
```

If you want to implement this in a different way that's fine but it looks like there is a gap in Figma between buttons.

**Review by brandyscarney — COMMENTED** [review]
Looks good besides the missing gap!

File: core/src/components/datetime/datetime.ionic.scss (hunk 3 above)
Comment (thetaPC) [reply core/src/components/datetime/datetime.ionic.scss]: https://github.com/ionic-team/ionic-framework/pull/30014/commits/0e0641950f1bbc95a15ff0d2214f39c5ea9ed2a8 and https://github.com/ionic-team/ionic-framework/pull/30014/commits/6d3fe984e42fe48037197c8359bf6206b0cdf5d6

**Pull request merged**